launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -498,6 +504,8 @@ protected boolean handle(String opt, String value) {
         case DRIVER_MEMORY -> conf.put(SparkLauncher.DRIVER_MEMORY, value);
         case DRIVER_JAVA_OPTIONS -> conf.put(SparkLauncher.DRIVER_EXTRA_JAVA_OPTIONS, value);
         case DRIVER_LIBRARY_PATH -> conf.put(SparkLauncher.DRIVER_EXTRA_LIBRARY_PATH, value);
+        case DRIVER_DEFAULT_CLASS_PATH ->
+          conf.put(SparkLauncher.DRIVER_DEFAULT_EXTRA_CLASS_PATH, value);
```

Comment:
Hmm, why spark-submit can only specify driver default extra class path, cannot it specify executor default extra class path?

Reply:
No~ This is `--driver-class-path` which used as a special case because the driver JVM is started already.

We already have more general ones are `spark.executor.extraClassPath` and `spark.driver.extraClassPath` for both driver and executor. This PR extends the above with the following.
```
spark.driver.defaultExtraClassPath
spark.executor.defaultExtraClassPath
```

Reply:
Please see here.
- https://spark.apache.org/docs/latest/configuration.html#runtime-environment

![Screenshot 2024-02-24 at 01 08 02](https://github.com/apache/spark/assets/9700541/a40a38bb-ed2b-4a2e-83c4-6b81fe4729e0)

Reply:
BTW, is your concern is about that the following is insufficient?
```
  private[spark] val EXECUTOR_CLASS_PATH =
    ConfigBuilder(SparkLauncher.EXECUTOR_EXTRA_CLASSPATH)
      .withPrepended(EXECUTOR_DEFAULT_EXTRA_CLASS_PATH.key, File.pathSeparator)
```

Reply:
Let me double-check it.

Reply:
To @viirya ,
- In general, `bin/spark-submit ... -c spark.driver.defaultExtraClassPath="" -c spark.executor.defaultExtraClassPath=""` is a way to launch without hive-jackson dependency because `hive-jackson/*` are provided by those configuration. This works for the drivers of `cluster` mode submission and executors of both `client` and `cluster` submission modes always.
- Apache Spark already provides `--driver-class-path` is used only for the cases where we cannot use `spark.driver.extraClassPath`. In this case, `spark.driver.defaultExtraClassPath` is also not applicable. So, this PR adds ` --driver-default-class-path ""` in the same way.

Reply:
This is identical with the way of the existing Apache Spark's `spark.*.extraClassPath` way.

Reply:
Yea, `-c spark.driver.defaultExtraClassPath=""` should be same as `--driver-default-class-path ""`, so I wonder why we need it especially? And also what's the case we cannot specify it through `-c spark.driver.defaultExtraClassPath`?

It's probably okay as we already have `--driver-class-path` so this addition follows it. Just a question if you know the exact reason.

Reply:
`--driver-class-path` is required because Spark's Driver JVMs are started before loading `Spark`'s property files or config file.

Reply:
We cannot create `SparkConf` class if we cannot load Spark's jar files. This is the chicken and egg situation. To solve this issue, we blindly load `jars/*`. So, if we have other jars, we need to use `--driver-class-path`